tests/Checker/AJAX_Runner_Tests.php

```diff
@@ -0,0 +1,145 @@
+<?php
+/**
+ * Tests for the Checks class.
+ *
+ * @package plugin-check
+ */
+
+use WordPress\Plugin_Check\Checker\AJAX_Runner;
+use WordPress\Plugin_Check\Checker\Check_Result;
+
+class AJAX_Runner_Tests extends WP_UnitTestCase {
+
+	public function test_is_plugin_check_returns_true() {
+		// Mock the AJAX request.
+		$_SERVER['REQUEST_URI'] = '/admin-ajax.php?action=plugin_check_run_checks';
+		$_REQUEST['action']     = 'plugin_check_run_checks';
+		$_REQUEST['_wpnonce']   = wp_create_nonce( 'plugin_check_run_checks' );
+
+		$runner = new AJAX_Runner();
+
+		$this->assertTrue( $runner->is_plugin_check() );
+	}
+
+	public function test_is_plugin_check_returns_false() {
+		$runner = new AJAX_Runner();
+
+		$this->assertFalse( $runner->is_plugin_check() );
+	}
```

Comment:
Would be good to also cover specifically the case where it's still an AJAX request, but the `action` is something else.